google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -502,7 +506,40 @@ public boolean delete(BlobId blob) {
 
   @Override
   public Blob compose(ComposeRequest composeRequest) {
-    return todo();
+    final Map<StorageRpc.Option, ?> optionsMap =
+        StorageImpl.optionMap(Iterables.toArray(composeRequest.getTargetOptions(), Option.class));
+    GrpcCallContext grpcCallContext = GrpcRequestMetadataSupport.create(optionsMap);
+    ComposeObjectRequest.Builder composeObjectReqBuilder = ComposeObjectRequest.newBuilder();
+    final List<SourceObject> sources =
+        Lists.newArrayListWithCapacity(composeRequest.getSourceBlobs().size());
+    for (ComposeRequest.SourceBlob sourceBlob : composeRequest.getSourceBlobs()) {
+      sources.add(
+          SourceObject.newBuilder()
```

Comment:
I did not think this was worth writing a codec but could be convinced otherwise.